src/diff_stats.c

```diff
@@ -300,13 +300,31 @@ int git_diff_stats_to_buf(
 
 	if (format & GIT_DIFF_STATS_FULL || format & GIT_DIFF_STATS_SHORT) {
 		error = git_buf_printf(
-			out, " %" PRIuZ " file%s changed, %" PRIuZ
-			" insertion%s(+), %" PRIuZ " deletion%s(-)\n",
-			stats->files_changed, stats->files_changed != 1 ? "s" : "",
-			stats->insertions, stats->insertions != 1 ? "s" : "",
-			stats->deletions, stats->deletions != 1 ? "s" : "");
+				out, " %" PRIuZ " file%s changed",
+				stats->files_changed, stats->files_changed != 1 ? "s" : "");
+		
+		if (error < 0) 
+			return error;
```

Comment:
You don't need to check for errors every time. You can keep writing into the buffer and do a `git_buf_oom(out)` check in the end.